app/controllers/projects_controller.rb

```diff
@@ -18,7 +18,10 @@ def get_projects
   # GET /projects/1
   # GET /projects/1.json
   def show
-    @project.increase_views(current_user)
+    if current_visit && !Ahoy::Event.find_by(visit_id: current_visit.id)
+      ahoy.track("Visited project " + @project.id.to_s)
```

Comment:
What does ahoy.track do ?

Reply:
Ahoy records **Visits** and **Events**. A **Visit** is recorded automatically whenever someone visits the site. It is persistent for 4 hours by default and recognises users. An **Event** corresponds to a user doing something on a site and is called with `ahoy.track`.  

An Event has a `name`, eg. "Viewed Home page" which is passed to  `ahoy.track`. The `visit_id` corresponding to the Visit on which the Event was created is one of the columns stored and I am using that to check if another Event was logged from the same visit, in which case, don't increment the counter.

Reply:
```ruby
  Ahoy::Event.exists?(visit_id: current_visit.id)
```